app/models/user_role.rb

```diff
@@ -36,6 +36,7 @@ class UserRole < ApplicationRecord
     manage_roles: (1 << 17),
     manage_user_access: (1 << 18),
     delete_user_data: (1 << 19),
+    bypass_invite_limits: (1 << 20),
```

Comment:
I think instead of adding a `bypass_invite_limits` permission, I would add a `restricted_invites` permission or similar, thus preserving the current meaning and not requiring a database migration.

Reply:
Thank you for reviewing this!

The idea behind this was since invites are enabled by default, restrictions should be applied by default too.
And as staff should be exempt, this unfortunately necessitates a migration.
Though in theory, I could invert this and add it to the default permission, but I think that would also affect staff.
So to clarify: Should this kept as a default or not?

Reply:
Oh, you are right, I forgot invites were handled as “default permissions” and that it was not explicitly set for moderators and admins but inherited from the default…

However, existing installs will already have a “Everyone” role that explicitly includes the invite permission, so no migration is required for them.

So I think a way forward that does not involve running a migration is the following:
- make the new permission flag be `restricted_invite_users` or something instead of `bypass_invite_limits` and have `invite_users` keep its existing scope
- change the default flags to be `restricted_invite_users` rather than `invite_users`
- add `invite_users` to `config/roles.yml`
- change `validate_dangerous_permissions` to allow both `invite_users` and `restricted_invite_users` (with e.g. a `Flags::UNPRIVILEGED` constant that would be a superset of `Flags::DEFAULT`)

Reply:
Sorry I haven't addressed any of these requested changes yet.
It's more involved than I'm happy with, but I'm currently working on it and it's mostly done, but I haven't tested it yet and I still need to rewrite the specs.
I'm not comfortable pushing unfinished work and I really need to restrict myself on how much I do, so I unfortunately can't give a timeframe.

Reply:
> The owner role is easy to fix by removing `restricted_invite_users` from `ALL`, but I wasn't able to find a save way to do that with the admin and moderator role.

Well, I didn't account for that roles can't give permissions they don't have themselves. So either the owner role can't create invites without limits being applied, or they can't set that permission for other roles.
Or I make yet another exemption.
At this point I'm basically refactoring how roles work, so honestly I would rather leave this as is.